### PR TITLE
[SYCL] Fix library destructor priority value

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

File: sycl/source/detail/global_handler.cpp [hunk 1]
```diff
@@ -142,9 +142,9 @@ BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpReserved) {
   return TRUE; // Successful DLL_PROCESS_ATTACH.
 }
 #else
-// Setting maximum priority on destructor ensures it runs after all other global
-// destructors.
-__attribute__((destructor(65535))) static void syclUnload() { shutdown(); }
+// Setting minimal priority on destructor ensures it runs after all other global
+// destructors. Priorities 0-100 are reserved by the compiler.
+__attribute__((destructor(101))) static void syclUnload() { shutdown(); }
```
Comment (romanovvlad) [inline sycl/source/detail/global_handler.cpp:147]: What about setting something like 110 to let other libraries to have a destructor which works after SYCL one.
@smaslov-intel @alexbatashev  Would do you think?

Comment (alexbatashev) [reply sycl/source/detail/global_handler.cpp:147]: Makes sense, I think

File: sycl/source/detail/global_handler.cpp [hunk 2]
```diff
@@ -142,9 +142,11 @@ BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpReserved) {
...
 }
 #else
-// Setting minimal priority on destructor ensures it runs after all other global
-// destructors. Priorities 0-100 are reserved by the compiler.
-__attribute__((destructor(101))) static void syclUnload() { shutdown(); }
+// Setting low priority on destructor ensures it runs after all other global
+// destructors. Priorities 0-100 are reserved by the compiler. The priority
+// value 110 allows SYCL users to run their destructors after runtime library
+// deinitialization.
+__attribute__((destructor(110))) static void syclUnload() { shutdown(); }
```
Comment (smaslov-intel) [inline sycl/source/detail/global_handler.cpp:149]: Should we document it somewhere what priority are for what? And how it interacts with no priority specified?

Comment (alexbatashev) [reply sycl/source/detail/global_handler.cpp:149]: I conducted a small experiment:
```c++
#include <iostream>

__attribute__((destructor(101))) static void syclUnload101() {
    std::cout << "101\n";
}

__attribute__((destructor(102))) static void syclUnload102() {
    std::cout << "102\n";
}

__attribute__((destructor(110))) static void syclUnload110() {
    std::cout << "110\n";
}

__attribute__((destructor(105))) static void syclUnload105() {
    std::cout << "105\n";
}

__attribute__((destructor)) static void syclUnload() {
    std::cout << "default\n";
}

int main() {
    std::cout << "Hi\n";
    return 0;
}
```

The result is the same for both GCC and Clang:
```
Hi
default
110
105
102
101
```

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**